### ISSUE TITLE
Documentation for Service::Error behavior should say it is protocol dependent

### ISSUE THREAD

**Issue by gituser-rs** [hyperium/hyper#3659]
**Version**
1.3.1

**Platform**
Ubuntu 23.10

**Description**
I am using hyper-util `auto` server. I return an error from service. Instead of sending GOAWAY and closing connection hyper sends RST_STREAM. I looked into [documentation](https://docs.rs/hyper/latest/hyper/service/trait.HttpService.html#associatedtype.Error) and it looks like the connection should closed, but it sends RST.

![image](https://github.com/hyperium/hyper/assets/150919540/59bf59a8-57f6-467a-8890-620c6db2dc56)

```
let mut builder = Builder::new(TokioExecutor::new());;

.............

let service = service_fn(move |request| {
 async move {
   println!("request");

   Err::<Response<String>, &str>("error")
 }
}

if let Err(err) = builder.serve_connection(stream, service).await {
    println!("{err}")
}
```

**Comment by seanmonstar** [comment]
Ah ok, yea, that's a problem with the docs. They should be updated to say that the behavior depends on the protocol version. It will abort the request however the protocol allows, either with some sort of RST_STREAM, or killing the connection if that doesn't exist.

**Comment by seanmonstar** [comment]
Closed by #3712
